### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 12

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#252] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152315517-0df427a5-da5e-4444-b783-10c5d371c3f0.png)

**Comment by LuginiAndrea** [comment]
Dividiamo le 3 possibile casistiche in base agli insiemi:
Caso 1
$$A \subset B$$
$$A \subset B \Longrightarrow A \cup B = B \Longrightarrow f^{-1}\left(A \cup B\right) = f^{-1}\left(B\right)$$
Nel primo caso l'ipotesi è confermata.
Caso 2:
$$A \cap B \neq \emptyset$$
$$A \cap B \neq \emptyset \Longrightarrow A \cup B = \left(A \cap \overline{B}\right) \cup \left(\overline{A} \cap B\right) \cup \left(A \cap B\right) \Longrightarrow f^{-1}\left(A \cup B\right) = f^{-1}\left(A \cap \overline{B}\right) \cup f^{-1}\left(\overline{A} \cap B\right) \cup f^{-1}\left(A \cap B\right)$$
In quanto $\left(X \cap \overline{Y}\right)$ è l'insieme di tutti gli elementi esclusivi di $X$, possiamo dedurre che $\left(X \cap \overline{Y}\right) \cup \left(X \cap Y\right) = X$
Di conseguenza $f^{-1}\left(A \cap \overline{B}\right) \cup f^{-1}\left(A \cap B\right) = f^{-1}\left(A\right)$, e la stessa cosa è vera anche per $B$.
Ergo $$f^{-1}\left(A \cup B\right) = f^{-1}\left(A \cap \overline{B}\right) \cup f^{-1}\left(\overline{A} \cap B\right) \cup f^{-1}\left(A \cap B\right) = f^{-1}\left(A\right) \cup f^{-1}\left(B\right)$$
Caso 3:
$$A \cap B = \emptyset$$ 
Questa condizione è dimostrabile in modo analogo al caso 2
### Update
Più semplice ragionare per casi:

- $\exists y \in f^{-1}\left(A \cup B\right) \Longrightarrow \exists x \in A \cup B \mid f^{-1}\left(x\right) = y \Longrightarrow \left(x \in A \Longrightarrow y \in f^{-1}\left(A\right), x \in B \textrm{ analogo }\right)$
- $\exists y \in f^{-1}\left(A\right) \cup f^{-1}\left(B\right) \Longrightarrow \exists x \in A \vee \exists x \in B \mid f^{-1}\left(x\right) = y \implies x \in A \cup B \implies y \in f^{-1}\left(A \cup B\right)$
